src/component/segments/SegmentList/SegmentList.tsx

```diff
@@ -0,0 +1,144 @@
+import { useContext, useState } from 'react';
+import {
+    Table,
+    TableBody,
+    TableCell,
+    TableHead,
+    TableRow,
+    Typography,
+} from '@material-ui/core';
+import AccessContext from 'contexts/AccessContext';
+import usePagination from 'hooks/usePagination';
+import { ADMIN } from 'component/providers/AccessProvider/permissions';
+import PaginateUI from 'component/common/PaginateUI/PaginateUI';
+import { SegmentListItem } from './SegmentListItem/SegmentListItem';
+import { ISegment } from 'interfaces/segment';
+import PermissionButton from 'component/common/PermissionButton/PermissionButton';
+import { useStyles } from './SegmentList.styles';
+import { useSegments } from 'hooks/api/getters/useSegments/useSegments';
+import { SegmentDeleteConfirm } from '../SegmentDeleteConfirm/SegmentDeleteConfirm';
+import { useSegmentsApi } from 'hooks/api/actions/useSegmentsApi/useSegmentsApi';
+import useToast from 'hooks/useToast';
+import { formatUnknownError } from 'utils/format-unknown-error';
+
+export const SegmentsList = () => {
+    const { hasAccess } = useContext(AccessContext);
+    const { segments, refetchSegments } = useSegments();
+    const { deleteSegment } = useSegmentsApi();
+    const { page, pages, nextPage, prevPage, setPageIndex, pageIndex } =
+        usePagination(segments, 10);
+    const [currentSegment, setCurrentSegment] = useState<ISegment | null>(null);
+    const [delDialog, setDelDialog] = useState(false);
+    const [confirmName, setConfirmName] = useState('');
+    const { setToastData, setToastApiError } = useToast();
+
+    const styles = useStyles();
+
+    const onDeleteSegment = async () => {
+        if (!currentSegment?.id) return;
+        try {
+            await deleteSegment(currentSegment?.id);
+            refetchSegments();
+            setToastData({
+                type: 'success',
+                title: 'Successfully deleted segment',
+                text: 'Your segment is now deleted',
+            });
+        } catch (error: unknown) {
+            setToastApiError(formatUnknownError(error));
+        }
+        setDelDialog(false);
+        setConfirmName('');
+    };
+
+    const renderSegments = () => {
+        return page.map((segment: ISegment) => {
+            return (
+                <SegmentListItem
+                    key={segment.id}
+                    id={segment.id}
+                    name={segment.name}
+                    description={segment.description}
+                    createdAt={segment.createdAt}
+                    createdBy={segment.createdBy}
+                    //@ts-expect-error
+                    setCurrentSegment={setCurrentSegment}
```

Comment:
We can drop the error annotation by changing the state to:

```ts
const [currentSegment, setCurrentSegment] = useState<ISegment>();
```

...and the prop to:

```ts
setCurrentSegment: React.Dispatch<React.SetStateAction<ISegment | undefined>>
```